Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -273,6 +291,13 @@ public class MapboxNavigationService: NSObject, NavigationService {
         
         poorGPSTimer.disarm()
         router.delegate = nil
+        
+        // Navigator should also be paused to prevent further location updates. In case if billing
+        // session is not running anymore - do nothing.
```

Comment:
Calling `NavigationService.stop()` several times, this can be considered as a developer error. `testNavigationServiceStartStopFinishSeveralTimes` tests this as well.